GameMap added

Heratexx opened the pull request:
fixes #4 

The GameMap need function from Tower, Player, Pausemenu and Inventory to display the different Components.

This Version of the GameMape displays placeholder for each component

Heratexx committed:
GameMap added

:exclamation: The GameMap need funtion from Tower, Player, Pausemenu and Inventory to Display the different components.

Heratexx committed:
Merge pull request #1 from SWE-C3/master

Merge fixes

Heratexx committed:
autopep8 fail fixed

after executing requirements.txt and requirements-win32.txt again, autopep8 could run without fails.

KnutZuidema commented:
Noch einmal vom Haupt Projekt pullen, dann gehen die checks auch durch
Edit: gehen sie nicht weil der fix dafür in einem anderen pull request ist. Alles gut hier

Heratexx committed:
Merge pull request #2 from SWE-C3/master

PullRequest#2

KnutZuidema reviewed (commented):
Bin mir nicht sicher wie das durch den Linter ging, aber Methoden sollten snake_case sein

KnutZuidema commented on GameMap.py:
```diff
@@ -0,0 +1,73 @@
+        map_size = map_window.getmaxyx()
+        # self.player.current_level.print
+        map_window.addstr(
+            int(map_size[0] / 2), int(map_size[1] / 2) - 7, "{current level}")
```
integer division funktioniert in python mit `//`, fände es gut wenn wir das verwenden

Heratexx replied:
Ah ok. Find ich persönlich etwas komisch zu lesen, aber damit es konsistent bleibt.

KnutZuidema commented on GameMap.py:
```diff
@@ -0,0 +1,73 @@
+        health_bar_window.refresh()
+
+    # Call Event based on key_input
+    def EventHandler(self, key_input):
```
Methoden sollten snake_case sein

Heratexx committed:
updated function names

function names from CamleCase to snake_case

KnutZuidema commented on src/utility.py:
```diff
@@ -20,4 +21,4 @@ def template(values):
         y_pos += 1
-    window.addstr(y_pos, x_pos, separator)
\ No newline at end of file
+    window.addstr(y_pos, x_pos, separator)
```
Bitte nicht die bestehenden Funktionen in der utility.py ändern, das ist im default Branch des Hauptprojekts schon geschehen und ich kann sonst nicht mergen

Heratexx replied:
Das hat der autopep durchlauf gemacht

KnutZuidema replied:
hmm ja dann mach das mal manuell wieder rückgängig. Hatte zu dem Zeitpunkt noch ein paar Probleme mit der CI und so 😅 Oder du updatest einfach das projekt nochmal vom hauptprojekt. Kann ich aber auch automatisch machen wenn du die Änderung an der utility.py wieder rückgängig machst

Heratexx replied:
ok das mache ich dann jetzt noch

KnutZuidema reviewed (commented):
Die Datei sollte nach snake_case benannt sein und in den Ordner `src` gelegt werden

Heratexx committed:
Small changes

Rename file (snake_case)
now using int division by using //

KnutZuidema commented on src/game_map.py:
```diff
@@ -0,0 +1,73 @@
+        # start with full life
+        health_bar = "||||||||||"
+        # cut one "|" off for each 10th between current life and full life (rounded off)
+        for index in range(0, 10 - int(health / 10)):
```
Hier bitte noch zu `//` wechseln

Heratexx committed:
Small change

Heratexx committed:
Merge branch 'master' of github.com:SWE-C3/Adventure-Game into SWE-C3-master

Heratexx commented:
Hoffe ich habs jetzt gefixt

KnutZuidema committed:
Merge branch 'master' into master

KnutZuidema reviewed (changes requested):
Hier nocheinmal der pylint output, jetzt wo das modul in stc liegt:
```
src/game_map.py:8:10: W0511: TODO: read events from file (?) (fixme)
src/game_map.py:36:0: C0301: Line too long (90/79) (line-too-long)
src/game_map.py:64:0: C0301: Line too long (88/79) (line-too-long)
src/game_map.py:1:0: C0111: Missing module docstring (missing-docstring)
src/game_map.py:4:0: C0111: Missing class docstring (missing-docstring)
src/game_map.py:18:4: C0111: Missing method docstring (missing-docstring)
src/game_map.py:52:4: C0111: Missing method docstring (missing-docstring)
src/game_map.py:56:11: R0123: Comparison to literal (literal-comparison)
src/game_map.py:61:4: C0111: Missing method docstring (missing-docstring)
src/game_map.py:61:4: R0201: Method could be a function (no-self-use)
src/game_map.py:71:4: C0111: Missing method docstring (missing-docstring)
```

Heratexx commented:
Muss ich die W0511 fixen ? das TODO wäre ja schon für jeden der an dem Kram weiter arbeiten will nützlich.

Heratexx committed:
pylint changes

changes

Heratexx committed:
Merge branch 'master' of github.com:Heratexx/Adventure-Game

Heratexx committed:
more changes

Heratexx committed:
event_handler changes

comment event_handler logic out.

KnutZuidema commented on src/game_map.py:
```diff
@@ -0,0 +1,94 @@
+        # get function from event dict
+        event = self.events.get(key_input, "no event")
+        # if there is any function
+        if event is not "no event":
```
`is` bzw. `is not` ist nicht das Gleiche wie `==` oder `!=`. `is` vergleicht ob die Instanz gleich ist, also:
```python
string = 'test'
string is string # -> True
string is 'test' # -> False
```

Heratexx replied:
Ah ok habe es jetzt erstmal auskommentiert, macht ja eh nichts. Passe das aber jetzt gleich direkt noch an.

Heratexx committed:
draw_health_bar changes

Heratexx committed:
changes

Heratexx commented:
************* Module src.game_map
src/game_map.py:73:4: R0201: Method could be a function (no-self-use) 

was mache ich dagegen am besten ? wenn wir den Spieler mal haben dann werde ich da self auch benutzen müssen, aber jetzt halt noch nicht

KnutZuidema commented:
> 
> 
> ************* Module src.game_map
> src/game_map.py:73:4: R0201: Method could be a function (no-self-use)
> 
> was mache ich dagegen am besten ? wenn wir den Spieler mal haben dann werde ich da self auch benutzen müssen, aber jetzt halt noch nicht

Statt `health` zu übergeben, benutze es in der Form `self.player.health`. So soll es ja später wahrscheinlich auch verwendet werden oder?

Heratexx committed:
draw_health_bar fixes

refactored draw_health_bar => update_health_bar
update_health_bar: no return value anymore
update_health_bar: updates the self.health_bar variable

Heratexx commented:
self.player.health gibt es ja noch nicht. Das sollte so doch nicht gehen oder ?

KnutZuidema commented:
Doch das würde gehen, da sowas in Python nur zur runtime gecheckt wird, aber so wie du es gelöst hast ist das auch ok

Heratexx commented:
Ah ok. Ich muss mich daran echt erst gewöhnen
